cdap-docs/build.sh

```diff
@@ -38,227 +66,197 @@ function set_project_path() {
+    build_docs_outer_level ${source_path} 
+    copy_docs_inner_level
+  fi
+  local warnings="$?"
+  if [ "${warnings}" != "0" ]; then
+    return ${warnings}
+  fi
+  if [ "${javadocs}" == "${WITH}" ]; then
+    run_command javadocs
+  fi
+  if [ "${doc_type}" == "${GITHUB}" -o "${doc_type}" == "${GITHUB_ONLY}" ]; then
+    run_command docs-github-part ${source_path} 
+  elif [ "${doc_type}" == "${WEB}" -o "${doc_type}" == "${WEB_ONLY}" ]; then
+    run_command docs-web-part ${source_path} 
   fi
-}
-  
-function copy_docs_lower_level() {
-  echo ""
-  echo "========================================================"
-  echo "Copying lower-level documentation..."
   echo "--------------------------------------------------------"
+  _bell "Completed build of \"${doc_type}\""
+  echo "========================================================"
+  echo "========================================================"
   echo ""
-
-  for i in ${MANUALS}; do
-    copy_html ${i}
-  done
-
-  local project_dir
-  # Rewrite 404 file, using branch if not a release
-  if [ "x${GIT_BRANCH_TYPE}" == "xfeature" ]; then
-    project_dir=${PROJECT_VERSION}-${GIT_BRANCH}
-  else
-    project_dir=${PROJECT_VERSION}
-  fi
-  rewrite ${BUILD}/${HTML}/404.html "src=\"_static"  "src=\"/cdap/${project_dir}/en/_static"
-  rewrite ${BUILD}/${HTML}/404.html "src=\"_images"  "src=\"/cdap/${project_dir}/en/_images"
-  rewrite ${BUILD}/${HTML}/404.html "/href=\"http/!s|href=\"|href=\"/cdap/${project_dir}/en/|g"
-  rewrite ${BUILD}/${HTML}/404.html "action=\"search.html"  "action=\"/cdap/${project_dir}/en/search.html"
+  exit 0
 }
 
-################################################## current
-
-function build_all() {
-  echo ""
-  echo "========================================================"
+function build_javadocs() {
   echo "========================================================"
-  echo "Building All: GitHub and Web Docs."
+  echo "Building Javadocs"
   echo "--------------------------------------------------------"
   echo ""
-  echo "========================================================"
-  echo "Building GitHub Docs."
+  build_javadocs_api
   echo "--------------------------------------------------------"
-  echo ""
-  run_command docs-github-part ${ARG_2} ${ARG_3}
-  echo "Stashing GitHub Docs."
-  cd ${SCRIPT_PATH}
-  mkdir -p ${SCRIPT_PATH}/${BUILD_TEMP}
-  mv ${SCRIPT_PATH}/${BUILD}/*.zip ${SCRIPT_PATH}/${BUILD_TEMP}
-  echo ""
+  echo "Completed Build of Javadocs"
   echo "========================================================"
-  echo "Building Web Docs."
-  echo "--------------------------------------------------------"
-  echo ""
-  run_command docs-web-part ${ARG_2} ${ARG_3}
   echo ""
+}
+
+function build_javadocs_api() {
+  set_mvn_environment
+  MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=128m" mvn clean install -P examples,templates,release -DskipTests -Dgpg.skip=true && mvn clean site -DskipTests -P templates -DisOffline=false
```

Comment:
Sorry, I meant the actual two `mvn` calls. Both have clean in them. Are we simply trying to fill the Maven cache with the first? Anything built will otherwise be removed by the clean in the next `mvn` command, so I am not sure why we're doing them both this way.